### PR TITLE
Allow to configure config.json location for kube-aggregator

### PR DESCRIPTION
**Pull request by mfojtik**
@deads2k PTAL

(I tried to run the image as a Pod but I can't mount a ConfigMap into same context as where the binary is... thus a need to configure different path ;-)

### CODE REVIEW

File: cmd/kube-aggregator/pkg/cmd/server/start.go [hunk 1]
```diff
@@ -43,14 +43,19 @@ import (
...
+	defaultConfigFile     = "config.json"
+)
 
 type DiscoveryServerOptions struct {
 	Etcd           *genericoptions.EtcdOptions
 	SecureServing  *genericoptions.SecureServingOptions
 	Authentication *genericoptions.DelegatingAuthenticationOptions
 	Authorization  *genericoptions.DelegatingAuthorizationOptions
 
+	ConfigFile string
```
Comment (sttts) [inline cmd/kube-aggregator/pkg/cmd/server/start.go]: nit: ConfigFilePath

**Comment by sttts** [comment]
lgtm. one nit.

**Comment by mfojtik** [comment]
@sttts nit fixed.

File: cmd/kube-aggregator/config.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
         "servers": [
                 {
-                        "serverAddress": "http://127.0.0.1:8083",
+                        "serverAddress": "https://openshift.default:443",
```
Comment (deads2k) [inline cmd/kube-aggregator/config.json]: Heh, no.  Also, do we use this for anything?

File: cmd/kube-aggregator/pkg/cmd/server/start.go [hunk 3]
```diff
@@ -172,9 +179,8 @@ func (o DiscoveryServerOptions) RunDiscoveryServer() error {
 
 // RunLegacyDiscoveryServer runs the legacy mode of discovery
 func (o DiscoveryServerOptions) RunLegacyDiscoveryServer() error {
-	configFilePath := "config.json"
 	port := "9090"
-	s, err := legacy.NewDiscoverySummarizer(configFilePath)
+	s, err := legacy.NewDiscoverySummarizer(o.ConfigFilePath)
```
Comment (deads2k) [inline cmd/kube-aggregator/pkg/cmd/server/start.go]: Oh, we got permission to straight up remove this.  Just delete the entire codepath.

File: cmd/kube-aggregator/config.json (hunk 2 above)
Comment (mfojtik) [inline cmd/kube-aggregator/config.json]: hups, it must sneaked in by accident :)

File: cmd/kube-aggregator/pkg/cmd/server/start.go (hunk 3 above)
Comment (sttts) [inline cmd/kube-aggregator/pkg/cmd/server/start.go]: 👍

File: cmd/kube-aggregator/config.json (hunk 2 above)
Comment (lavalamp) [inline cmd/kube-aggregator/config.json]: lol!

File: cmd/kube-aggregator/pkg/cmd/server/start.go [hunk 4]
```diff
@@ -43,14 +43,19 @@ import (
 	"k8s.io/kubernetes/cmd/kube-aggregator/pkg/apis/apiregistration/v1alpha1"
 )
 
-const defaultEtcdPathPrefix = "/registry/kubernetes.io/kube-aggregator"
+const (
+	defaultEtcdPathPrefix = "/registry/kubernetes.io/kube-aggregator"
```
Comment (lavalamp) [inline cmd/kube-aggregator/pkg/cmd/server/start.go]: apologies for the drive-by, but I'd recommend changing the default to not start with registry--that way there's less chance of it and the main apiserver fighting if they both happen to get pointed at the same etcd. Maybe a different PR. @deads2k

File: cmd/kube-aggregator/pkg/cmd/server/start.go [hunk 5]
```diff
@@ -93,6 +99,7 @@ func NewCommandStartDiscoveryServer(out, err io.Writer) *cobra.Command {
 	o.Authorization.AddFlags(flags)
 	flags.StringVar(&o.ProxyClientCertFile, "proxy-client-cert-file", o.ProxyClientCertFile, "client certificate used identify the proxy to the API server")
 	flags.StringVar(&o.ProxyClientKeyFile, "proxy-client-key-file", o.ProxyClientKeyFile, "client certificate key used identify the proxy to the API server")
+	flags.StringVar(&o.ConfigFilePath, "config-file-path", o.ConfigFilePath, "path to a config file")
```
Comment (lavalamp) [inline cmd/kube-aggregator/pkg/cmd/server/start.go]: Document this better, where should a user look to find the contents of this config file?

File: cmd/kube-aggregator/pkg/cmd/server/start.go [hunk 6]
```diff
@@ -43,14 +43,19 @@ import (
 	"k8s.io/kubernetes/cmd/kube-aggregator/pkg/apis/apiregistration/v1alpha1"
 )
 
-const defaultEtcdPathPrefix = "/registry/kubernetes.io/kube-aggregator"
+const (
+	defaultEtcdPathPrefix = "/registry/kubernetes.io/kube-aggregator"
+	defaultConfigFilePath = "config.json"
```
Comment (lavalamp) [inline cmd/kube-aggregator/pkg/cmd/server/start.go]: Can we default to .yaml? (more human readable?)

File: cmd/kube-aggregator/pkg/cmd/server/start.go (hunk 4 above)
Comment (deads2k) [inline cmd/kube-aggregator/pkg/cmd/server/start.go]: > apologies for the drive-by, but I'd recommend changing the default to not start with registry--that way there's less chance of it and the main apiserver fighting if they both happen to get pointed at the same etcd. Maybe a different PR. @deads2k

I'll open a pull to move it to `/registry/kube-aggregator.kubernetes.io`

Comment (deads2k) [inline cmd/kube-aggregator/pkg/cmd/server/start.go]: Updated in https://github.com/kubernetes/kubernetes/pull/41031

**Comment by deads2k** [comment]
This is obsolete.  The code has been deleted.
